src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -545,22 +545,6 @@ private bool Equals(TupleTypeSymbol other, bool ignoreCustomModifiers, bool igno
                 return false;
             }
 
-            // Make sure field names are the same.
-            if (!ignoreDynamic)
```

Comment:
I think we need to keep this. There is code in CreateConversion that relies on compares with ignoreDynamic == false when it needs to figure whether we have a conversion that does not change anything (including dynamic and names) or there is some change. In the latter case it needs to ensure that there is a conversion node in the tree for the purpose of semantic model. 
I.E. if we have a conversion that changes only names, we that code must put something in the tree and it relies on equality to not match for original and converted type when there is a change in names or dynamic.